open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/buffergraph/BufferGraphConnector.java

```diff
@@ -72,124 +60,129 @@ public void setMainGraph(Object mainGraph) {
+                if (vertexToStart != null) {
+                    columnOut = findPathForOutputAsset(vertexToStart, g, startingVertex);
 
                 }
 
-                moveColumnProcessColumn(columnIn,columnOut,process);
+                moveColumnProcessColumn(columnIn, columnOut, process);
             }
         }
 
     }
 
-    private void moveColumnProcessColumn(Vertex columnIn,Iterator<Vertex> columnOut,Vertex process){
+    private void moveColumnProcessColumn(Vertex columnIn, Iterator<Vertex> columnOut, Vertex process) {
         if (columnOut != null && columnOut.hasNext()) {
             String columnOutGuid = columnOut.next().values(PROPERTY_KEY_ENTITY_GUID).next().toString();
             String columnInGuid = columnIn.values(PROPERTY_KEY_ENTITY_GUID).next().toString();
             if (!columnOutGuid.isEmpty() && !columnInGuid.isEmpty()) {
-                MainGraphMapper mainGraphMapper = new MainGraphMapper(bufferGraph,mainGraph);
-                mainGraphMapper.checkBufferGraph(columnInGuid,columnOutGuid,process);
+                MainGraphMapper mainGraphMapper = new MainGraphMapper(bufferGraph, mainGraph);
+                mainGraphMapper.checkBufferGraph(columnInGuid, columnOutGuid, process);
             }
         }
     }
 
-    private void addVerticesAndRelationship(GraphTraversalSource g, GraphContext nodeToNode){
+    private void addVerticesAndRelationship(GraphTraversalSource g, GraphContext nodeToNode) throws JanusConnectorException {
         LineageEntity fromEntity = nodeToNode.getFromVertex();
         LineageEntity toEntity = nodeToNode.getToVertex();
 
-        Vertex vertexFrom = addVertex(g,fromEntity);
-        Vertex vertexTo = addVertex(g,toEntity);
+        Vertex vertexFrom = addVertex(g, fromEntity);
+        Vertex vertexTo = addVertex(g, toEntity);
 
         //add check gia null vertex
-        addRelationship(nodeToNode.getRelationshipGuid(),nodeToNode.getRelationshipType(),vertexFrom,vertexTo);
+        addRelationship(nodeToNode.getRelationshipGuid(), nodeToNode.getRelationshipType(), vertexFrom, vertexTo);
 
     }
 
-    private  Vertex addVertex(GraphTraversalSource g,LineageEntity lineageEntity){
+    private Vertex addVertex(GraphTraversalSource g, LineageEntity lineageEntity) throws JanusConnectorException {
         final String methodName = "addVertex";
 
         Iterator<Vertex> vertexIt = g.V().has(PROPERTY_KEY_ENTITY_GUID, lineageEntity.getGuid());
         Vertex vertex;
 
-        if(!vertexIt.hasNext()){
+        if (!vertexIt.hasNext()) {
             vertex = g.addV(lineageEntity.getTypeDefName()).next();
-            addPropertiesToVertex(g,vertex,lineageEntity);
+            addPropertiesToVertex(g, vertex, lineageEntity);
             g.tx().commit();
-        }
-        else {
+        } else {
             vertex = vertexIt.next();
             log.debug("{} found existing vertex {}", methodName, vertex);
```

Comment:
specially for debug logging we use to have 
```
if (log.isDebugEnabled()) {
    log.debug("{} found existing vertex {}", methodName, vertex);
}
```
to avoid formatting the string even debugging is not enabled, and message is not logged, which might have high cost.